packages/modal/src/useModal.ts

```diff
@@ -9,18 +9,37 @@ import { useState } from 'react';
 import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
 import { useFocusJail } from '@zendeskgarden/container-focusjail';
 
-export function useModal({ onClose, modalRef, id: _id, focusOnMount, environment } = {}) {
+export interface IUseModalProps {
+  onClose?: (event: Event) => any;
```

Comment:
Do we actually return an `event` with `onClose`? This may need to be removed.

`onClose` also doesn't return any value so this could be `void`

Reply:
If we accept 2 values this could be `event: MouseEvent | KeyboardEvent`

Reply:
Updated return type to be `void` and event to be a union of `MouseEvent` and `KeyboardEvent` types.